plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/connection/ConnectionWizardPageModel.java

```diff
@@ -176,60 +205,99 @@ public void createConnection() {
-					resetConnection(ValidationStatus.error(NLS.bind("Could not connect to host at {0}.", host), e));
-				}
+				updateModel(null, 
+						factory, 
+						ValidationStatus.error(NLS.bind("Host at {0} is not an {1} host.", host, factory.getName())), 
+						Status.OK_STATUS);
 			}
+		} catch (IOException e) {
+			updateModel(null, 
+					factory, 
+					ValidationStatus.error(NLS.bind("Could not connect to host at {0}.", host), e), 
+					Status.OK_STATUS);
 		}
+	}
 
-		setConnection(connection);
+	private void createConnection(String host, IConnectionsFactory connectionsFactory) {
+		IConnection connection;
+		try {
+			connection = connectionsFactory.create(host);
+			if (connection != null) {
+				updateModel(connection, 
+						connectionFactory, 
+						Status.OK_STATUS,
+						Status.OK_STATUS);
+			} else {
+				updateModel(null, 
+						connectionFactory,
+						ValidationStatus.error(NLS.bind("The host at {0} is no OpenShift host", host)),
+						Status.OK_STATUS);
+			}
+		} catch (IOException e) {
+			updateModel(null, 
+					AutomaticConnectionFactoryMarker.getInstance(),
+					ValidationStatus.error(NLS.bind("Could not connect to host at {0}.", host), e), 
+					Status.OK_STATUS);
+		}
+	}
+
+	private boolean equalsTypeAndHost(IConnection thisConnection, IConnection thatConnection) {
+		if ((thisConnection == null && thatConnection != null)
+				|| (thatConnection == null && thisConnection != null)) {
+			return false;
+		} else if (thisConnection == null && thatConnection == null) {
+			return true;
+		}
+		return thisConnection.getType().equals(thatConnection.getType())
+				&& thisConnection.getHost().equals(thatConnection.getHost());
 	}
 	
 	public void setUseDefaultHost(boolean useDefaultHost) {
-		firePropertyChange(PROPERTY_USE_DEFAULT_HOST, this.useDefaultHost, this.useDefaultHost = useDefaultHost);
-		if (useDefaultHost
-				&& connectionFactory != null) {
-			setHost(connectionFactory.getDefaultHost());
+		setUseDefaultHost(useDefaultHost, true);
+	}
+	
+	private void setUseDefaultHost(boolean useDefaultHost, boolean updateModel) {
+		if (updateModel) {
+			updateModel(null, connectionFactory, Status.CANCEL_STATUS, Status.OK_STATUS);
 		}
-		resetConnectionCreated();
+		firePropertyChange(PROPERTY_USE_DEFAULT_HOST, this.useDefaultHost, this.useDefaultHost = useDefaultHost);
 	}
 	
 	public boolean isUseDefaultHost() {
 		return useDefaultHost;
 	}
 
-	private void resetConnectionCreated() {
-		setConnectionCreated(null);
-	}
-
-	private void setConnectionCreated(IStatus status) {
-		firePropertyChange(PROPERTY_CONNECTION_CREATED, this.connectionCreated, this.connectionCreated = status);
+	private void setConnectionCreationError(IStatus status) {
+		firePropertyChange(PROPERTY_CONNECTION_CREATION_ERROR, this.connectionCreationError, this.connectionCreationError = status);
 	}
 
-	public IStatus getConnectionCreated() {
-		return connectionCreated;
+	public IStatus getConnectionCreationError() {
```

Comment:
yeah this only stores an error, only meaningful if the connection in this null (could not create it) so that we have a hint for the reason of the failure